packages/flutter_tools/lib/src/compile.dart

```diff
@@ -653,7 +653,7 @@ class ResidentCompiler {
     if (!_compileRequestNeedsConfirmation) {
       return Future<CompilerOutput>.value(null);
     }
-    _stdoutHandler.reset();
```

Comment:
A reject request doesn't have sources, so we hang indefinitely if this isn't reset